src/shuffle.rs

```diff
@@ -3,6 +3,21 @@ use rand::Rng;
 use crate::prng::make_prng;
 
 /// Shuffles a vector using the Fisher-Yates algorithm
+///
+/// ## Example
+///
+/// Shuffle a vector of integers
+///
+/// ```
+/// use nois::shuffle;
+///
+/// // We are randomly shuffling a vector of integers [1,2,3,4]
+/// # let mut data = vec![1, 2, 3, 4];
```

Comment:
You can use # to hide code from the example. This code is still compiled when you compile and run the examples as part of `cargo test --doc`.

I think the randomness can be hidden but the other code should be visible.

```suggestion
/// let mut data = vec![1, 2, 3, 4];
```